File: leetcode2025/2025_07/leetcode2025-07-25.cpp
#include <iostream>
#include <vector>
#include <climits>
#include <algorithm>
using namespace std;

class Solution {
public:
    int maxSum(vector<int>& nums) {
        sort(nums.begin(), nums.end());
        int res = 0, last = INT_MAX;
        for (int& i : nums){
            if (i > 0 && i != last){
                res += i;
                last = i;
            }
        }
        return nums.back()<0 ? nums.back() : res;
    }
};

int main(){
    Solution s;
    vector<int> nums = {1,2,3,4,5};
    cout << s.maxSum(nums) << endl;
}

File: leetcode2025/2025_07/leetcode2025-07-25.py
from typing import List

class Solution:
    def maxSum(self, nums: List[int]) -> int:
        nums.sort()
        res, last = 0, float("inf")
        for i in nums:
            if i > 0 and i != last:
                res += i
                last = i
        return res if nums[-1]>=0 else nums[-1]

        # if max(nums) < 0:
        #     return max(nums)
        # nums = list(set(nums))
        # res = 0
        # for i in nums:
        #     if i>0:
        #         res += i
        # return res

if __name__ == "__main__":
    s = Solution()
    nums = [1,2,3,4,5]
    print(s.maxSum(nums))
